consensus/core/src/api/error.rs

```diff
@@ -0,0 +1,25 @@
+use hashes::Hash;
+use thiserror::Error;
+
+use crate::tx::{TransactionId, TransactionOutpoint};
+
+#[derive(Error, Debug)]
+pub enum ConsensusError {
```

Comment:
I prefer to expose the underlying rule errors directly rather than converting them